### PR TITLE
Custom id generation PoC

### PR DESCRIPTION
**Pull request by aberezkin**
Just an example of what can be a solution to #591. If this looks right I'll proceed to add it to other places where `nanoid` is currently used.

### CODE REVIEW

File: pnpm-lock.yaml [hunk 1]
```diff
@@ -46,7 +46,7 @@ importers:
         version: 0.6.2
       ai:
         specifier: 2.2.12
-        version: link:../../packages/core
+        version: 2.2.12(react@18.2.0)(solid-js@1.7.7)(svelte@4.0.1)(vue@3.3.4)
```
Comment (shuding) [inline pnpm-lock.yaml:49]: We might need to revert this change.

**Review by shuding — COMMENTED** [review]
Looks good to me! Would you mind copying the implementation to other framework APIs as well (Svelte, Vue, Solid)? Thanks!

File: pnpm-lock.yaml (hunk 1 above)
Comment (aberezkin) [reply pnpm-lock.yaml:49]: Yeah, don't know why it's even there to be honest. Maybe I have a different set package manager setup. I'll try to figure it up.

Comment (MaxLeiter) [reply pnpm-lock.yaml:49]: @aberezkin if you can fix the merge conflic tin `use-chat` and re-run `pnpm i` from the root directory this should be resolved and we can merge this PR.

**Comment by aberezkin** [comment]
Uh oh, I rebased and force pushed and the PR has closed. Will open another one with addressed comments.

**Comment by aberezkin** [comment]
Here's the new pull request 🤗

https://github.com/vercel/ai/pull/733

**Pull request closed without merging**